src/snake/main.rs

```diff
@@ -0,0 +1,470 @@
+extern crate termion;
+
+use termion::{IntoRawMode, TermWrite, Color, async_stdin};
+use std::io::{stdout, stdin, Read, Write};
+use std::time::{Instant, Duration};
+use std::collections::VecDeque;
+
+#[derive(PartialEq, Copy, Clone)]
+enum Direction {
+    Up,
+    Down,
+    Left,
+    Right,
+}
+
+/// Snake's Body Part location and direction
+struct BodyPart {
+    x: u16,
+    y: u16,
+    direction: Direction,
+}
+
+/// Snake's Food
+struct Food {
+    x: u16,
+    y: u16,
+}
+
+impl BodyPart {
+    fn crawl(&self) -> BodyPart {
+        let mut x = self.x;
+        let mut y = self.y;
+        let new_direction: Direction;
+
+        match self.direction {
+            Direction::Up => { 
+                y += 1;
+                new_direction = Direction::Up;
+            },
+            Direction::Down => { 
+                y -= 1;
+                new_direction = Direction::Down;
+            },
+            Direction::Left => {
+                x -= 1;
+                new_direction = Direction::Left;
+            },
+            Direction::Right => {
+                x += 1;
+                new_direction = Direction::Right;
+            },
+        };
+
+        BodyPart {
+            x: x,
+            y: y,
+            direction: new_direction
+        }
+    }
+}
+
+/// Snake
+struct Snake {
+    direction: Direction,
+    body: VecDeque<BodyPart>,
+}
+
+/// The game state.
+struct Game<R, W> {
+    /// The play area width.
+    width: usize,
+    /// The play area height.
+    height: usize,
+    /// Standard input.
+    stdin: R,
+    /// Standard output.
+    stdout: W,
+    /// Snake
+    snake: Snake,
+    /// Snake's Food
+    food: Food,
+    /// Speed
+    speed: u64,
+    /// Game Score
+    score: i32,
+    /// Interval between frames
+    interval: u64,
```

Comment:
Is it really necessary to store this in the struct? Also, why not use `Duration`, instead?